fix: custom installation of DOMINO

tristan-f-r opened the pull request:
[this has the added surprise that the pypi version of DOMINO is different than the one on GitHub!]

This installs DOMINO locally, which closes https://github.com/Reed-CompBio/spras/issues/137, by using local paths instead of hard-coded static paths, which originally caused the issue when running in Singularity.

This patches DOMINO with two patch files that separate runner.py (since before the behavior was reliant on setup.py) and move module imports outside of `src`.

This also makes a new change that allows for required inputs to have custom extensions. 

- [x] Depends on #226 (because of documentation)

tristan-f-r committed:
fix: custom installation of DOMINO

[this has the added surprise that the pypi version of DOMINO is different than the one on GitHub!]

This installs DOMINO locally, resolving https://github.com/Reed-CompBio/spras/issues/137. However, I would like to tackle #119 and #234, and I'm still missing test cases to show the resolution of #137.

This patches DOMINO with two patch files that separate runner.py (since before the behavior was reliant on setup.py) and move module imports outside of `src`.

tristan-f-r committed:
refactor: prefer path

tristan-f-r commented:
I'll try that out!

tristan-f-r committed:
test: add egfr test case

this works!

tristan-f-r commented:
That was a great test case - I got a new `DOMINO` error (most likely pertaining to the new changes in this PR), which I will investigate.

ntalluri commented on test/DOMINO/input/domino-active-genes-egfr.txt:
@agitter would using the egfr data as a test case cause the egfr data to be apart of the pilot data?

tristan-f-r replied:
I tried using a random graph earlier but that didn't work - however, since I know the size of a dataset to use, I can try to artificially induce node hotspots in a randomly generated graph and trick DOMINO into believing it is a biological network, which would work as a test case (as this isn't a test of "is DOMINO a good AMI algorithm," but rather "does DOMINO even run" - none of our test cases before actually made DOMINO run.

[To properly make the test cases work, I would need to read `stderr` but the stderr parsing is in #220 - for now, the tests require manual verification that its output does not include errors, which, locally for me, it doesn't.]

tristan-f-r committed:
test: drop psuedonodes, put nodes in ensemble format

also adjusts the docker container to move patch copies after requirements installation

tristan-f-r committed:
fix: finalize snakemake support for inputs of different file exts

tristan-f-r committed:
fix: unlink .sif.pkl, specify proper params in network

tristan-f-r commented:
The tests for this worked locally, but that was because the generate inputs test suite never cleaned up its inputs before running the test. Pushing a commit to fix that :+1:

tristan-f-r committed:
refactor: move extend_filename to util, generate_inputs respects ext

tristan-f-r committed:
Merge branch 'main' into domino-fix

tristan-f-r committed:
Merge branch 'main' into domino-fix

tristan-f-r committed:
Merge branch 'main' into domino-fix

tristan-f-r committed:
chore: mv to sif

tristan-f-r committed:
Merge branch 'main' into domino-fix

tristan-f-r committed:
Merge remote-tracking branch 'upstream/main' into domino-fix

tristan-f-r committed:
chore(typos): ignore sif

tristan-f-r committed:
Merge branch 'main' into domino-fix

tristan-f-r committed:
fix: drop snakemake pythonic file transform, use correct input

tristan-f-r committed:
Merge branch 'main' into domino-fix

tristan-f-r committed:
style: fmt

tristan-f-r committed:
fix: correctly import os

tristan-f-r committed:
Merge branch 'main' into domino-fix
